scripts/staging-reset.js

```diff
@@ -0,0 +1,114 @@
+const NEW_STAGING_BRANCH = `staging-${NEW_STAGING_NUMBER}`
+
+async function main() {
+  await initGitConfig()
+  await executeCommand(`git checkout ${MAIN_BRANCH} -f`)
+  await executeCommand(`git pull`)
+
+  const isNewBranch = CURRENT_STAGING_BRANCH !== NEW_STAGING_BRANCH
+  if (isNewBranch) {
+    console.log(`Changing staging branch in ${CI_FILE}...`)
+    await replace({
+      files: CI_FILE,
+      from: /CURRENT_STAGING: staging-.*/g,
+      to: `CURRENT_STAGING: ${NEW_STAGING_BRANCH}`,
+    })
+    await executeCommand(`git commit ${CI_FILE} -m "👷 Bump staging to ${NEW_STAGING_BRANCH}"`)
+    await executeCommand(`git push origin ${MAIN_BRANCH}`)
+  } else {
+    console.log(`Staging branch already up to date in ${CI_FILE}. Skipping.`)
+  }
+
+  const isStagingAlreadyCreated = await executeCommand(`git ls-remote --heads ${REPOSITORY} ${NEW_STAGING_BRANCH}`)
+  if (!isStagingAlreadyCreated) {
+    console.log(`Creating the new staging branch...`)
+    await executeCommand(`git checkout -b ${NEW_STAGING_BRANCH}`)
+    await executeCommand(`git push origin ${NEW_STAGING_BRANCH}`)
+  } else {
+    console.log(`New staging branch already created. Skipping.`)
+  }
+
+  await executeCommand(`git checkout ${CURRENT_STAGING_BRANCH}`)
+  if (isNewBranch && fs.existsSync(CI_FILE)) {
+    console.log(`Disabling CI on the old branch...`)
+    await executeCommand(`git rm ${CI_FILE}`)
+    await executeCommand(`git commit ${CI_FILE} -m "Remove ${CI_FILE} on old branch so pushes are noop"`)
+    await executeCommand(`git push origin ${CURRENT_STAGING_BRANCH}`)
+  } else {
+    console.log(`CI already disabled on the old branch. Skipping.`)
+  }
+
+  console.log('Reset Done.')
+}
+
+async function initGitConfig() {
+  const GITHUB_DEPLOY_KEY = await getSecretKey('ci.browser-sdk.github_deploy_key')
+
+  await executeCommand(`ssh-add - <<< "${GITHUB_DEPLOY_KEY}"`)
+  await executeCommand(`mkdir -p ~/.ssh`)
+  await executeCommand(`chmod 700 ~/.ssh`)
+  await executeCommand(`ssh-keyscan -H github.com >> ~/.ssh/known_hosts`)
+  await executeCommand(`git config user.email "browser-sdk-staging-reset@datadoghq.com"`)
+  await executeCommand(`git config user.name "Gitlab staging reset job"`)
+  await executeCommand(`git remote set-url origin ${REPOSITORY}`)
+}
+
+function getSecretKey(name) {
+  const awsParameters = [
+    'ssm',
+    'get-parameter',
+    `--region=us-east-1`,
+    '--with-decryption',
+    '--query=Parameter.Value',
+    '--out=text',
+    `--name=${name}`,
+  ]
+
+  return execute(`aws ${awsParameters.join(' ')}`, false)
+}
+
+function getIsoWeekNumber() {
+  const today = new Date()
+  const todayUtc = new Date(Date.UTC(today.getFullYear(), today.getMonth(), today.getDate()))
+  const dayNum = todayUtc.getUTCDay() || 7
+  todayUtc.setUTCDate(todayUtc.getUTCDate() + 4 - dayNum)
+  const yearStart = new Date(Date.UTC(todayUtc.getUTCFullYear(), 0, 1))
+  return Math.ceil(((todayUtc - yearStart) / 86400000 + 1) / 7)
```

Comment:
`web-ui` defines this function like so:

```
const getWeekNumber = (today = new Date()) => {
    const jan1 = new Date(today.getUTCFullYear(), 0, 1);
    return Math.ceil(((today - jan1) / 86400000 + jan1.getUTCDay() + 1) / 7);
};
```

Your implementation is a bit more complex, and I'm not sure which is better, do you have some clues here? I suggest to use the same logic for both projects.

In particular, the `+ 4` in your implementation looks suspicious.